include/avif/avif_cxx.h

```diff
@@ -0,0 +1,35 @@
+// Copyright 2023 Google LLC
+// SPDX-License-Identifier: BSD-2-Clause
+
+// This is a header only library.
+
+#ifndef AVIF_AVIF_CXX_H
+#define AVIF_AVIF_CXX_H
+
+#if !(defined(__cplusplus) || defined(c_plusplus))
```

Comment:
We should not test the `c_plusplus` macro. I guess that's a nonstandard macro defined by an old compiler.

Another way to see this is that this should be the opposite of the test in the C header:

```
#ifdef __cplusplus
extern "C" {
#endif
```

Note: I think this check is not necessary because compilation will fail if this header is included by C files. I guess you want a more informative error message?

Reply:
I removed c_plusplus
Exactly: it might as well fail with a clear message.